src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -474,13 +474,28 @@ private bool UpdateCacheSizeExceedsCapacity(CacheEntry entry, CoherentState cohe
             return true;
         }
 
+        private int lockFlag;
```

Comment:
Does the flag need to be volatile? I'm not sure with the memory model.

Reply:
Seems to me that it doesn't, since we access it through `Interlocked`.  And when we write to it, there is no need in it being volatile since according to [this](https://igoro.com/archive/volatile-keyword-in-c-memory-model-explained/) we write operations are "kind of volatile" and we will see that change anyway.

Reply:
And we can also follow the examaple from here: https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked.exchange?view=net-8.0

And replace `lockFlag = 0;` with Interlocked.Exchange.

Reply:
Also the last example provided here: https://github.com/dotnet/runtime/blob/main/docs/design/specs/Memory-model.md

<img width="773" alt="image" src="https://github.com/dotnet/runtime/assets/107400703/f702a17c-c225-4370-99bd-95893adbcfb5">


Shown that all that we need is a volatile read (in our case `Interlocked.CompareExchange`) and you can have non volatile write and you will still see the changes eventually.